mlflow/utils/environment.py

```diff
@@ -193,7 +193,8 @@ def _is_iterable(x):
 
 
 _INFER_PIP_REQUIREMENTS_FALLBACK_MESSAGE = (
-    "Encountered an unexpected error while inferring pip requirements (model URI: %s, flavor: %s)"
+    "Encountered an unexpected error while inferring pip requirements (model URI: %s, flavor: %s)."
+    " To disable the auto dependency inference, specify `conda_env` or `pip_requirements`."
 )
```

Comment:
This change assumes that `_infer_pip_requirements` is called from a `save_model()` / `log_model()` function, but users might call it from somewhere else. Do we need to make this change, since every flavor has a fallback?